rest_api/schema.py

```diff
@@ -23,16 +23,17 @@ class FilterRequest(BaseModel):
     filters: Optional[Dict[str, Optional[Union[str, List[str]]]]] = None
 
 
-
 @pydantic_dataclass
 class AnswerSerialized(Answer):
     context: Optional[str] = None
 
+
 @pydantic_dataclass
 class DocumentSerialized(Document):
     content: str
     embedding: Optional[List[float]]
 
+
 @pydantic_dataclass
 class LabelSerialized(Label):
```

Comment:
You mentioned that we need to send an id here. Totally agree that this shouldn't be mandatory for the client.  Wonder if a simple `id = Optional[str]` is enough here as we already support some auto-init in the class here: https://github.com/deepset-ai/haystack/blob/84147edcca7c3d9cb510ac39d861533e0112c803/haystack/schema.py#L318 

If this doesn't work let's not tackle it in this PR, but can you please create an issue for this so that we don't forget about it?

Reply:
I definitely tried that, but it rejected it with a bizzare error (something about `id` being an Optional that was preceding a mandatory arg, regardless of where I put it in the params order), so after a while I gave up :sweat_smile: